api/lock_api.go

```diff
@@ -144,15 +142,8 @@ type Committer struct {
 	Email string `json:"email"`
 }
 
-// CurrentCommitter returns a Committer instance populated with the same
-// credentials as would be used to author a commit. In particular, the
-// "user.name" and "user.email" configuration values are used from the
-// config.Config singleton.
-func CurrentCommitter() Committer {
```

Comment:
I'm fine with keeping a dependency on the `config` here. What do you think about:

```go
func CommitterFromConfiguration(c *config.Configuration) Committer {
        return Committer{
                Name: c.Git.Get("user.name"),
                Email: c.Git.Get("user.email")
        }
}
```

or... using the config unmarshalling stuff like:

```go
type Committer struct {
        Name  string `git:"user.name"`
        Email string `git:"user.email"`
}

func NewCommitter(cfg *config.Configuration) Committer {
        var c Committer
        if err := cfg.Unmarshal(&c); err != nil {
                // Should never happen, a `panic` is appropriate
                // here.
                panic(err.Error())
        }

       return c
}

As an added benefit, this allows us to get rid of some usage of the `api.CurrentCommitter()` func.